Adyen/Analytics/AnalyticsProvider/AnalyticsProvider.swift

```diff
@@ -32,16 +32,26 @@ public struct AnalyticsConfiguration {
 public struct AdditionalAnalyticsFields {
     /// The amount of the payment
     public let amount: Amount?
+    
+    public let sessionId: String?
+    
+    public init(amount: Amount? = nil, sessionId: String? = nil) {
+        self.amount = amount
+        self.sessionId = sessionId
+    }
 }
 
 @_spi(AdyenInternal)
-public protocol AnalyticsProviderProtocol: TelemetryTrackerProtocol {
+public protocol InitialTelemetryProtocol {
     
-    var checkoutAttemptId: String? { get }
-    
-    func fetchAndCacheCheckoutAttemptIdIfNeeded()
+    /// Sends the initial data and retrieves the checkout attempt id as a response.
+    func fetchCheckoutAttemptId(with flavor: TelemetryFlavor, additionalFields: AdditionalAnalyticsFields?)
+}
+
+@_spi(AdyenInternal)
+public protocol AnalyticsProviderProtocol: InitialTelemetryProtocol {
     
-    var additionalFields: (() -> AdditionalAnalyticsFields)? { get }
+    var checkoutAttemptId: String? { get }
```

Comment:
Do we need to create `InitialTelemetryProtocol` and `AnalyticsProviderProtocol` protocols? Can't we add  `checkoutAttemptId` in `InitialTelemetryProtocol`?

Reply:
Hmm, there may not be a reason for them to stay separate any more. Will update if that's the case, nice catch :)